File: Ejercicio 3/ordenar_vector.js
const readline = require('readline');

/**
 * Ordena un vector de números de menor a mayor
 * @param {number[]} numeros - Array de números a ordenar
 * @returns {number[]} - Array ordenado
 */
function ordenarVector(numeros) {
    return numeros.slice().sort((a, b) => a - b);
}

/**
 * Función principal del programa
 */
async function main() {
    const rl = readline.createInterface({
        input: process.stdin,
        output: process.stdout
    });

    console.log("=== ORDENAR VECTOR DE NÚMEROS ===");
    console.log();

    try {
        // Solicitar al usuario que ingrese los números
        console.log("Ingresa los números separados por espacios (ejemplo: 5 2 8 1 9):");
        
        const entrada = await new Promise((resolve) => {
            rl.question("Números: ", resolve);
        });

        // Convertir la entrada en un array de números
        const numeros = entrada.split(' ').map(x => parseFloat(x)).filter(x => !isNaN(x));

        if (numeros.length === 0) {
            console.log("Error: No se ingresaron números válidos.");
            rl.close();
            return;
        }

        console.log(`\nVector original: [${numeros.join(', ')}]`);

        // Ordenar el vector
        const vectorOrdenado = ordenarVector(numeros);

        console.log(`Vector ordenado: [${vectorOrdenado.join(', ')}]`);

        // Mostrar estadísticas adicionales
        const suma = vectorOrdenado.reduce((acc, num) => acc + num, 0);
        const promedio = suma / vectorOrdenado.length;

        console.log(`\nEstadísticas:`);
        console.log(`- Cantidad de elementos: ${vectorOrdenado.length}`);
        console.log(`- Número más pequeño: ${Math.min(...vectorOrdenado)}`);
        console.log(`- Número más grande: ${Math.max(...vectorOrdenado)}`);
        console.log(`- Promedio: ${promedio.toFixed(2)}`);

    } catch (error) {
        console.log(`Error inesperado: ${error.message}`);
    } finally {
        rl.close();
    }
}

// Ejecutar el programa
if (require.main === module) {
    main();
} 

File: Ejercicio 3/ordenar_vector.py
def ordenar_vector(numeros):
    """
    Ordena un vector de números de menor a mayor
    """
    return sorted(numeros)

def main():
    print("=== ORDENAR VECTOR DE NÚMEROS ===")
    print()
    
    # Solicitar al usuario que ingrese los números
    print("Ingresa los números separados por espacios (ejemplo: 5 2 8 1 9):")
    entrada = input("Números: ")
    
    try:
        # Convertir la entrada en una lista de números
        numeros = [float(x) for x in entrada.split()]
        
        if not numeros:
            print("Error: No se ingresaron números.")
            return
        
        print(f"\nVector original: {numeros}")
        
        # Ordenar el vector
        vector_ordenado = ordenar_vector(numeros)
        
        print(f"Vector ordenado: {vector_ordenado}")
        
        # Mostrar estadísticas adicionales
        print(f"\nEstadísticas:")
        print(f"- Cantidad de elementos: {len(vector_ordenado)}")
        print(f"- Número más pequeño: {min(vector_ordenado)}")
        print(f"- Número más grande: {max(vector_ordenado)}")
        print(f"- Promedio: {sum(vector_ordenado) / len(vector_ordenado):.2f}")
        
    except ValueError:
        print("Error: Asegúrate de ingresar solo números válidos separados por espacios.")
    except Exception as e:
        print(f"Error inesperado: {e}")

if __name__ == "__main__":
    main() 